content/blog/2022/headed-playwright-in-docker.md

```diff
@@ -16,56 +16,69 @@ summary: |
   interface from outside Docker containers.
 ---
 
-[Playwright](https://playwright.dev/) is a test runner that uses real browsers
-to test web applications (an alternative to tools like
-[Selenium](https://www.selenium.dev/)). By default, Playwright runs these
+[Playwright] is a test runner that uses real browsers to test web applications
+(an alternative to tools like [Selenium]). By default, Playwright runs these
 browsers in headless mode, which means the pages are loaded and tested without
 opening the browser window. This is great when running entire test suites
 locally, and in CI where having a bunch of rapidly opening windows would be
-disconcerting. However, when it comes to writing or debugging individual
-tests, it is convenient to open the browser in *headed* mode to actually see the
-page being tested.
-
-Playwright includes the [Playwright
-Inspector](https://playwright.dev/docs/debug#playwright-inspector) to
-conveniently launch the browser in headed mode *and* a separate window to
-control test execution. The inspector is launched by setting `PWDEBUG=1` before
-calling `playwright`, and is compatible with all browsers supported by
-Playwright (Safari, Chrome, and Firefox) in all major operating systems.
-Playwright also includes a `playwright open <url>` subcommand to quickly launch
-the inspector on any URL. Examples in this post use `playwright open <url>`, but
-`PWDEBUG=1` is equivalent.
-
-Playwright also provides a convenient [Docker
-image](https://playwright.dev/docs/docker), which includes all three browsers
-pre-installed and configured so you can skip the dependency installation steps.
-But what happens if you try to run the Playwright Inspector inside a Docker
-container -- which normally doesn't have a graphical user interface?
+disconcerting. However, when it comes to writing or debugging individual tests,
+it is convenient to open the browser in _headed_ mode to actually see the page
+being tested.
+
+[Playwright]: https://playwright.dev/
+[Selenium]: https://www.selenium.dev/
+
+Playwright for JS includes [UI mode] to let you run end-to-end tests with a
+graphical explorer and debugger, allowing you to view, run, and debug tests in a
+real browser window. For other supported platforms (like Python), Playwright
```

Comment:
I don't feel strongly, but I wonder if this could be in a `{% callout 'note' %}` so that it's clear you've made updates?